python/demo/gsoc2022/utils.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+from ufl import as_vector
+from dolfinx import io
+from mpi4py import MPI
+
+
+def save_as_xdmf(filepath, mesh, E):
+
+    with io.XDMFFile(MPI.COMM_WORLD, filepath, "w") as xdmf:
+        xdmf.write_mesh(mesh)
+        xdmf.write_function(E)
+
+
+class background_electric_field:
+
+    def __init__(self, theta, n_bkg, k0):
+        self.theta = theta
+        self.k0 = k0
+        self.n_bkg = n_bkg
+
+    def eval(self, x):
+
+        kx = self.n_bkg * self.k0 * np.cos(self.theta)
+        ky = self.n_bkg * self.k0 * np.sin(self.theta)
+        phi = kx * x[0] + ky * x[1]
+
+        ax = np.sin(self.theta)
+        ay = np.cos(self.theta)
+
+        return (-ax * np.exp(1j * phi), ay * np.exp(1j * phi))
+
+
+def radial_distance(x):
+    return np.sqrt(x[0]**2 + x[1]**2)
+
+
+def curl_2d(a):
```

Comment:
Not sure about having these in a separate file, typically our demos are in one file minus perhaps lengthy mesh generation steps.